linera-service/src/node_service.rs

```diff
@@ -80,10 +80,10 @@ enum NodeServiceError {
     ChainClientError(#[from] ChainClientError),
     #[error(transparent)]
     BcsHexError(#[from] BcsHexParseError),
-    #[error("could not decode query string")]
+    #[error("could not decode query string: {0}")]
     QueryStringError(#[from] hex::FromHexError),
     #[error(transparent)]
-    BcsError(#[from] bcs::Error),
+    Bcs(#[from] bcs::Error),
```

Comment:
Then we could also remove the `Error` suffixes from the other variants.

Reply:
From all of them?
We have plenty: `ChainError`, `JsonError`, `RemoteNodeError`, `LocalNodeError`, `OverWeightError`, `ViewError`, `ArithmeticError`, and so on.
I am fine with this, I would just want to avoid a misunderstanding.

Reply:
In my opinion yes: If an enum is called `SomethingFoo` then any suffix `Foo` in any of its variants is redundant. (See also [the `suffix` Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/suffix), which doesn't _quite_ apply here.)

But maybe that's controversial, so we could discuss with the team first. Not a blocker from my side.

Reply:
I think I agree, _unless_ the error variant is a wrapper around another error type. Then the variant name should be the same as the nested error type's name IMHO. But yes, we should probably discuss with the team first 😅

Reply:
I'm onboard with the strong form of this, including removing `Error` from wrapper variants.  I'm not really sure what the `Error` suffix adds to the name here.